### ISSUE TITLE
HEAD fails to build on macOS with stock `bison`

### ISSUE THREAD

**Issue by iamitava** [doxygen/doxygen#7091]
I'm on macOS Mojave (10.14.5).  A fresh checkout of the master branch fails to build with this error:

```
[ 31%] [BISON][constexp] Building parser with bison 2.3
constexp.y:43.14-25: syntax error, unexpected string, expecting =
make[2]: *** [generated_src/ce_parse.cpp] Error 1
make[1]: *** [src/CMakeFiles/_doxygen.dir/all] Error 2
make: *** [all] Error 2
```

The default `bison` on macOS is version 2.3.  a18e967 removed the `=` on that line (`%name-prefix="constexpYY"`) with the comment that it's deprecated since 2.3b.

Even after restoring the `=`, there's another error on the next line (`%define api.pure full`):

```
[ 31%] [BISON][constexp] Building parser with bison 2.3
constexp.y:44.9-16: syntax error, unexpected identifier, expecting string
make[2]: *** [generated_src/ce_parse.cpp] Error 1
make[1]: *** [src/CMakeFiles/_doxygen.dir/all] Error 2
make: *** [all] Error 2
```

This is part of the changes from 8412f4d.  I believe "%define api.pure" was also introduced in 2.3b, deprecating "%pure-parser".

Please consider adding a version requirement for bison in `CMakeLists.txt`.  `cmake` doesn't seem to understand a version requirement such as `find_package(BISON 2.3b REQUIRED)` so I am not sure how to do that.

Meanwhile, reverting to the older syntax works:

```patch
diff --git a/src/constexp.y b/src/constexp.y
index 62a51f33..80d7e2d9 100644
--- a/src/constexp.y
+++ b/src/constexp.y
@@ -40,8 +40,8 @@ int constexpYYerror(yyscan_t yyscanner, const char *s)
 
 %}
 
-%name-prefix "constexpYY"
-%define api.pure full
+%name-prefix="constexpYY"
+%pure-parser
 %lex-param {yyscan_t yyscanner}
 %parse-param {yyscan_t yyscanner}
 
```

**Comment by albert-github** [comment]
As you already wrote doxygen uses some newer features that are not supported by the version that is by default installed on macOS Mojave (10.14.5). Easiest is to upgrade your bison version e.g. from `bincrafters` or through `brew`.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #7092

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of a release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.16.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
